### ISSUE TITLE
Invalid model

### ISSUE THREAD

**Issue by LeXXik** [playcanvas/editor#378]
Under some conditions, the Editor shows incorrect model in the scene view. In the following screenshot, the model should be a cylinder, but instead it is a bowling pin:

![image](https://user-images.githubusercontent.com/5677782/116062483-9c216a80-a68c-11eb-94ea-0bb0dd8fbbee.png)

Added @yaustar as admin to project (private due to model IP):
https://playcanvas.com/project/788485/overview/editor-model-issue

Edit:
The model shows correctly as a cylinder at runtime.

**Comment by LeXXik** [comment]
Additional finding related to this issue. The antity with a model component will show incorrect model, if it has a GLB asset picked in its Asset type property, even if Cylinder type is selected.

**Comment by willeastcott** [comment]
Interesting. So are you able to reproduce this outside of your private project yet, @LeXXik?

**Comment by LeXXik** [comment]
Yes, I noticed it today in another project (private as well, unfortunately), which allowed to find the reason. I will see if I can create a public reproducible.

**Comment by LeXXik** [comment]
Here is a repro. Additional info: one of the parents has to have a mesh collision component set to that model.
https://playcanvas.com/project/803068/overview/model-issue

**Comment by willeastcott** [comment]
This appears to be an engine bug related to `ModelComponent` (and `RenderComponent` too!). When changing the type property from 'asset' to a primitive type (like 'box'), they're not unbinding from the asset's event listeners. This meant that if the asset fired events later (particularly a 'load' event), it would replace the primitive model/mesh instances with the asset's content. Because the assets are loaded async in the Viewport, the bug shows there, but in the Launch page, assets are preloaded, so the load event happens before, not after the `type` property is changed. Assigning to @mvaligursky since this is engine related.

**Comment by mvaligursky** [comment]
So how do you reproduce it using this project?
I tried changing the model asset to be box instead of glb asset and back multiple times, with and without browser caching, and that seems to work reliably for me in both the Editor and the Launcher.
